kiwixbuild/dependencies/apple_xcframework.py

```diff
@@ -0,0 +1,98 @@
+import os
+import shutil
+from pathlib import Path
+
+from kiwixbuild.platforms import PlatformInfo
+from kiwixbuild.utils import pj, run_command
+from .base import Dependency, NoopSource, Builder as BaseBuilder
+
+
+class AppleXCFramework(Dependency):
+    all_targets = ["macOS_x86_64", "macOS_arm64_static", "iOS_x86_64", "iOS_arm64"]
```

Comment:
Renamed that all but I can't move it completely to AppleStaticAll otherwise I'd have circular dependencies. We still need class-level access inside the builder for get_dependencies.